browser/web_discovery/web_discovery_tab_helper.cc

```diff
@@ -41,13 +50,24 @@ void WebDiscoveryTabHelper::MaybeCreateForWebContents(
 
 WebDiscoveryTabHelper::WebDiscoveryTabHelper(content::WebContents* contents)
     : content::WebContentsObserver(contents),
-      content::WebContentsUserData<WebDiscoveryTabHelper>(*contents) {}
+      content::WebContentsUserData<WebDiscoveryTabHelper>(*contents) {
+#if BUILDFLAG(ENABLE_WEB_DISCOVERY_NATIVE)
+  if (base::FeatureList::IsEnabled(features::kBraveWebDiscoveryNative)) {
+    web_discovery_service_ = WebDiscoveryServiceFactory::GetForBrowserContext(
```

Comment:
if there is no particular reason to cache the service, it's better to obtain it straight at the point of use aka `MaybeExtractFromPage`

Reply:
why is this better?